tests/remote/pi_resources.go

```diff
@@ -0,0 +1,290 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package remote
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/IBM-Cloud/power-go-client/clients/instance"
+	"github.com/IBM-Cloud/power-go-client/ibmpisession"
+	"github.com/IBM-Cloud/power-go-client/power/models"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/klog/v2"
+)
+
+var (
+	piSession *ibmpisession.IBMPISession
+	piID      string
+)
+
+func (r *Remote) createPVSResources() (err error) {
+	var image, network string
+
+	piSession, err = geIBMPISession()
+	if err != nil {
+		return fmt.Errorf("error while creating ibm pi session: %v", err)
+	}
+
+	ic := instance.NewIBMPIInstanceClient(context.Background(), piSession, piID)
+	imgc := instance.NewIBMPIImageClient(context.Background(), piSession, piID)
+	nc := instance.NewIBMPINetworkClient(context.Background(), piSession, piID)
+	kc := instance.NewIBMPIKeyClient(context.Background(), piSession, piID)
```

Comment:
Please see whether you want to group these clients under a single struct like a powervsclients and map it to Remote obj, so that you can reuse them during destroy also you don't need to pass individual clients to each func.

Reply:
Updated code.